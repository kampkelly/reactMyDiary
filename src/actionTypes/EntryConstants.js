export const NEW_ENTRY = 'NEW_ENTRY';
