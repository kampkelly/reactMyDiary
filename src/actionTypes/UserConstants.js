export const SIGNIN = 'SIGNIN';
export const SIGNUP = 'SIGNUP';
export const SIGNOUT = 'SIGNOUT';
export const PROFILE = 'PROFILE';
export const NOTIFICATION = 'NOTIFICATION';