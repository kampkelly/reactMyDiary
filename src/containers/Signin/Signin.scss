main {
  // background-image: url("https://i.imgur.com/Xxvt7y4.jpg");
  background-size: cover;
  background-position: center;
  background-repeat-x: no-repeat;
  background-repeat-y: no-repeat;
  min-height: 100vh;
}
main h2 {
  margin-top: 0px;
  padding-top: 20px;
}

#signup h1 {
  font-size: 2.2em;
  margin-top: 0px;
  margin-bottom: 0px;
  padding-top: 10px;
  color: #DFAC2C;
}
#signup h5 {
  margin-bottom: 0px;
}
#signup .form-cover {
  background: rgba(107, 101, 101, 0.3);
  width: 61%;
  margin: auto;
}
#signup .submit-button {
  width: 100%;
}
@media (min-width: 0px) and (max-width: 599px) {
  #signup .form-cover {
    width: 90%;
    margin: auto;
  }
  #signup .submit-button {
    width: 100%;
  }
}
@media (min-width: 600px) and (max-width: 900px) {
  #signup .form-cover {
    width: 80%;
  }
  #signup .submit-button {
    width: 100%;
  }
  #signup [class*='col-'] {
    padding-right: 0;
  }
}
