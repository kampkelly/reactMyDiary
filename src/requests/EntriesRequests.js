import axios from 'axios';

import { asyncActions } from '../util/AsyncUtil';
import { ALL_ENTRIES } from '../actionTypes/EntryConstants';
import { entryConstant } from '../constants/Constants';

export const PaginatedEntries = limit => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}?limit=${limit}`)
    .then((response) => {
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const AllEntries = () => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}`)
    .then((response) => {
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};
