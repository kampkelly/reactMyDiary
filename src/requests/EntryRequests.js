import axios from 'axios';
import { createBrowserHistory } from 'history';

import { asyncActions } from '../util/AsyncUtil';
import { NEW_ENTRY, SHOW_ENTRY, UPDATE_ENTRY, ALL_ENTRIES, PAGINATED_ENTRIES, DELETE_ENTRY } from '../actionTypes/EntryConstants';
import { entryConstant } from '../constants/Constants';

const history = createBrowserHistory({ forceRefresh: true });

export const AddEntry = (title, description) => (dispatch) => {
  axios.post(`${entryConstant.ENTRIES_URL}`, { title, description })
    .then((response) => {
      dispatch(asyncActions(NEW_ENTRY).loading(false));
      dispatch(asyncActions(NEW_ENTRY).success(true));
      if (response.data.status === 'Success') {
        document.getElementById('loading').style.display = 'none';
        history.push(`/entries/${response.data.entry.id}?notice=${response.data.message}`);
      } else {
        document.getElementById('loading').style.display = 'none';
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
      dispatch(asyncActions(NEW_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(NEW_ENTRY)
      .failure(true, error.response.data.message)));
};

export const ShowEntry = id => (dispatch) => {
  // document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src="http://res.cloudinary.com/ddfepbdqg/image/upload/v1543597369/Rolling.svg" id="loading"></img>`)
  axios.get(`${entryConstant.ENTRIES_URL}/${id}`)
    .then((response) => {
      dispatch(asyncActions(SHOW_ENTRY).loading(false));
      document.getElementById('loading').style.display = 'none';
      if (response.data.status === 'Success') {
        dispatch(asyncActions(SHOW_ENTRY).success(response.data.entry));
      }
      dispatch(asyncActions(SHOW_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(SHOW_ENTRY)
      .failure(true, error.response.data.message)));
};

export const ModifyEntry = (id, title, description) => (dispatch) => {
  axios.put(`${entryConstant.ENTRIES_URL}/${id}`, { title, description })
    .then((response) => {
      dispatch(asyncActions(UPDATE_ENTRY).loading(false));
      if (response.data.status === 'Success') {
        dispatch(asyncActions(UPDATE_ENTRY).success(response.data.entry));
        document.getElementById('loading').style.display = 'none';
        history.push(`/entries/${response.data.entry.id}?notice=${response.data.message}`);
      } else {
        document.getElementById('loading').style.display = 'none';
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
      dispatch(asyncActions(UPDATE_ENTRY).loading(true));
    })
    .catch(error => dispatch(asyncActions(UPDATE_ENTRY)
      .failure(true, error.response.data.message)));
};

export const AllEntries = () => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}`)
    .then((response) => {
      document.getElementById('loading').style.display = 'none';
      dispatch(asyncActions(ALL_ENTRIES).loading(false));
      dispatch(asyncActions(ALL_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(ALL_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(ALL_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const PaginatedEntries = limit => (dispatch) => {
  axios.get(`${entryConstant.ENTRIES_URL}?limit=${limit}`)
    .then((response) => {
      dispatch(asyncActions(PAGINATED_ENTRIES).loading(false));
      dispatch(asyncActions(PAGINATED_ENTRIES).success(response.data.entries));
      dispatch(asyncActions(PAGINATED_ENTRIES).loading(true));
    })
    .catch(error => dispatch(asyncActions(PAGINATED_ENTRIES)
      .failure(true, error.response.data.message)));
};

export const DeleteEntry = id => (dispatch) => {
  axios.delete(`${entryConstant.ENTRIES_URL}/${id}`)
    .then((response) => {
      dispatch(asyncActions(DELETE_ENTRY).loading(false));
      dispatch(asyncActions(DELETE_ENTRY).success(true));
      dispatch(asyncActions(DELETE_ENTRY).loading(true));
      history.push('/dashboard?notice=Entry has been deleted');
    })
    .catch(error => dispatch(asyncActions(DELETE_ENTRY)
      .failure(true, error.response.data.message)));
};