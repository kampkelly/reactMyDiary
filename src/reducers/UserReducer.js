import { SIGNIN, SIGNUP, SIGNOUT, PROFILE, NOTIFICATION, UPDATE_PROFILE } from '../actionTypes/UserConstants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  isAuth: false,
  loading: false,
  message: '',
  success: false,
  failure: false,
  user: {}
};

const UserReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(SIGNIN).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SIGNIN).success:
      return {
        ...state,
        success: true, ...action.payload.user, isAuth: true
      };
    case asyncActionName(SIGNIN).failure:
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(SIGNUP).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SIGNUP).success:
      return {
        ...state,
        success: action.payload,
      };
    case asyncActionName(SIGNUP).failure:
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(PROFILE).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(PROFILE).success:
      return {
        ...state,
        success: true, user: action.payload
      };
    case asyncActionName(PROFILE).failure:
      return {
        ...state,
      };
    case asyncActionName(UPDATE_PROFILE).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(UPDATE_PROFILE).success:
      return {
        ...state,
        message: action.payload.message, success: true, user: action.payload.user
      };
    case asyncActionName(UPDATE_PROFILE).failure:
      document.querySelector('.form_error_text').style.display = 'block';
			document.querySelector('.form_error_text small').textContent = data.message;
      return {
        ...state,
      };
    case asyncActionName(NOTIFICATION).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(NOTIFICATION).success:
      return {
        ...state,
        success: true, message: action.payload.message, user: action.payload.user
      };
    case asyncActionName(NOTIFICATION).failure:
      return {
        ...state,
      };
    case asyncActionName(SIGNOUT).success:
      return initialState;
    default:
      return state;
  }
};

export default userReducer;
