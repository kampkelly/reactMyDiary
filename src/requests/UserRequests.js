import axios from 'axios';
import { createBrowserHistory } from 'history';

import setAuthToken from '../util/AuthTokenUtil';
import { asyncActions } from '../util/AsyncUtil';
import { SIGNIN, SIGNUP } from '../actionTypes/UserConstants';
import { userConstant } from '../constants/Constants';

const history = createBrowserHistory({ forceRefresh: true });

export const SigninUser = (email, password) => (dispatch) => {
  axios.post(userConstant.SIGNIN_URL, { email, password })
    .then((response) => {
      dispatch(asyncActions(SIGNIN).success(response.data));
      document.querySelector('.form_error_text').style.display = 'none';
      if (response.data.status === 'Success') {
        localStorage.setItem('diary_token', response.data.token);
        setAuthToken(response.data.token);
        history.push('/signin');
      } else {
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
    })
    .catch(error => dispatch(asyncActions(SIGNIN)
      .failure(true, error.response.data.message)));
};

export const SignupUser = (email, password, confirmPassword, dateOfBirth, fullName) => (dispatch) => {
  axios.post(userConstant.SIGNUP_URL, { email, password, confirmPassword, dateOfBirth, fullName })
    .then((response) => {
      dispatch(asyncActions(SIGNUP).success(response.data));
      document.querySelector('.form_error_text').style.display = 'none';
      if (response.data.status === 'Success') {
        localStorage.setItem('diary_token', response.data.token);
        setAuthToken(response.data.token);
        history.push('/signin');
      } else {
        document.querySelector('.form_error_text').style.display = 'block';
        document.querySelector('.form_error_text small').textContent = response.data.message;
      }
    })
    .catch(error => dispatch(asyncActions(SIGNUP)
      .failure(true, error.response.data.message)));
};
