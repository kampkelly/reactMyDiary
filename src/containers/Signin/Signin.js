import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './Signin.scss';
import '../../styles/form.scss';
import { SigninUser } from '../../requests/UserRequests';
import { validateForm } from '../../helpers/validateForm';

/**
 * @class Signin
 * @extends {Component}
 */
class Signin extends Component {
  /**
   *Creates an instance of Signin.
   * @memberof Signin
   */
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      success: false
    };
    this.submitSignin = this.submitSignin.bind(this);
    this.change = this.change.bind(this);
    this.goToHome = this.goToHome.bind(this);
  }

  /**
   * @description - This method runs whenever redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   * @memberof Header
   */
  static getDerivedStateFromProps(props, state) {
    return {
      failure: props.failure,
      success: props.success,
    };
  }

  change(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  goToHome() {
    this.props.history.push('/signin');
  }

  /**
   *
   *
   * @memberof Signin
   */
  submitSignin(e) {
    const errorMessage = 'One or more of the required fields are empty!';
    const values = { email: this.state.email, password: this.state.password };
    const noneEmpty = validateForm(errorMessage, values, e);
    if (noneEmpty === true) {
      // validation was successful
      const email = this.state.email.toLowerCase().replace(/\s+/g, '');
      const password = this.state.password.toLowerCase();
      this.props.SigninUser(email, password);
      document.querySelector('body').insertAdjacentHTML('afterbegin', `<img src="http://res.cloudinary.com/ddfepbdqg/image/upload/v1543597369/Rolling.svg" id="loading"></img>`);
    }
  }

  render() {
    const { email, password } = this.state;
    return (
      <div>
        <section id="signup">
          <h2 className="text-center white-text">Signin To Your Account</h2>
          <div className="blue-forms form-cover text-center">
            <form className="signin_form" id="signin-form">
              <p className="danger-text form_error_text"><small></small></p>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Email Address</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <input type="email" className="control-form" id="signup_email" name="email" value={email} onChange={this.change} autoComplete="email" />
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <label className="white-text">Password</label>
                </div>
                <div className="col-4-6 col-6-6-md col-6-6-xs">
                  <input type="password" className="control-form" id="signup_password" name="password" value={password} onChange={this.change} autoComplete="password" />
                </div>
              </div>
              <div className="field grid-container">
                <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                <div className="col-2-6 col-6-6-md col-6-6-xs">
                  <button type="submit" className="submit-button button-white" onClick={this.submitSignin}>Signin</button>
                </div>
                <div className="col-2-6 col-6-6-md col-6-6-xs mt-4-md mt-4-xs">
                  <button type="submit" className="submit-button  button-cancel mt-2-xs" onClick={this.goToHome}>Cancel</button>
                </div>
              </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <Link key="1" to="/signup" className="underline">Signup</Link>
                    <Link key="2" to="/forgot/password" className="underline">Forgot password?</Link>
                  </div>
                </div>
            </form>
        </div>
      </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  success: state.user.success
});

Signin.propTypes = {
  SigninUser: PropTypes.func,
  success: PropTypes.bool
};

export default connect(mapStateToProps, {
  SigninUser
})(Signin);
