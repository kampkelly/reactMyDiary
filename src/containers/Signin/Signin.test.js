import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Signin from './Signin';
import UserReducer from '../../reducers/UserReducer';
import { asyncActions } from '../../util/AsyncUtil';
import { SIGNIN } from '../../actionTypes/UserConstants';
import { SigninUser } from '../../requests/UserRequests';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  user: {
    user: {},
    message: '',
    success: false
  },
  entry: {
    entries: []
  }
});

const props = {
  history: {
    push: jest.fn()
  },
  submitSignin: jest.fn(),
};

let component;
let myComponent;

describe('<Signin/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <Signin {...props} />
      </Provider>
    );
    document.body.innerHTML =
    '<main></main>';
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should show profile', () => {
    const payload = {};
    const expectedAction = {
      type: 'SIGNIN_SUCCESS',
      payload
    };
    expect(asyncActions(SIGNIN).success(payload)).toEqual(expectedAction);
  });
  it('should dispatch an action when signin is loading', () => {
    const payload = false;
    const action = {
      type: 'SIGNIN_LOADING',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({
      loading: payload
    });
  });
  it('should dispatch an action to signin a user', () => {
    const payload = {
      message: '',
      user: {}
    };
    const action = {
      type: 'SIGNIN_SUCCESS',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({
      isAuth: true,
      success: true,
    });
  });
  it('should dispatch an action when signin a user fails', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const payload = {};
    const action = {
      type: 'SIGNIN_FAILING',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({});
  });
  it('should signin user', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const input = myComponent.find('input[name="email"]');
    input.simulate('change', { target: { value: 'email@example.com' } });
    myComponent.find('button.button-white').simulate('click', {
      preventDefault: () => {
      }
     });
  });
  it('should click cancel button', () => {
    myComponent.find('button.button-cancel').simulate('click');
  });
});
