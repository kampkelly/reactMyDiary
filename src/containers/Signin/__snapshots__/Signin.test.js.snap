// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin/> should render without throwing an error 1`] = `
<Connect(Signin)
  history={
    Object {
      "push": [MockFunction],
    }
  }
  submitSignin={[MockFunction]}
/>
`;
